src/test/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/mapper/induction/FutureWorkInterestsResourceMapperTest.kt

```diff
@@ -13,34 +16,140 @@ class FutureWorkInterestsResourceMapperTest {
+          role = "Kotlin Developer",
+        ),
+      ),
+    )
+    val expectedInterests = listOf(
+      WorkInterest(
+        workType = WorkInterestType.OTHER,
+        workTypeOther = "Entertainment",
+        role = "Juggler",
+      ),
+      WorkInterest(
+        workType = WorkInterestType.TECHNICAL,
+        workTypeOther = null,
+        role = "Kotlin Developer",
+      ),
+    )
 
     // When
     val actual = mapper.toUpdateFutureWorkInterestsDto(request, prisonId)
 
     // Then
     assertThat(actual!!.prisonId).isEqualTo(prisonId)
     assertThat(actual.reference).isEqualTo(request.id)
-    assertThat(actual.interests).hasSize(1)
-    assertThat(actual.interests[0].workType).isEqualTo(WorkInterestType.OTHER)
-    assertThat(actual.interests[0].workTypeOther).isEqualTo("Any job I can get")
-    assertThat(actual.interests[0].role).isEqualTo("Any role")
+    assertThat(actual.interests).usingRecursiveComparison().ignoringCollectionOrder().isEqualTo(expectedInterests)
+  }
+
+  @Test
+  fun `should map to UpdateFutureWorkInterestsDto with missing particular job interest`() {
+    // Given
+    val prisonId = "BXI"
+    val request = aValidUpdateWorkInterestsRequest(
+      workInterests = setOf(WorkType.OTHER, WorkType.TECHNICAL),
+      workInterestsOther = "Entertainment",
+      particularJobInterests = setOf(
+        WorkInterestDetail(
+          workInterest = WorkType.OTHER,
+          role = "Juggler",
+        ),
+      ),
+    )
+    val expectedInterests = listOf(
+      WorkInterest(
+        workType = WorkInterestType.OTHER,
+        workTypeOther = "Entertainment",
+        role = "Juggler",
+      ),
+      WorkInterest(
+        workType = WorkInterestType.TECHNICAL,
+        workTypeOther = null,
+        role = null,
+      ),
+    )
+
+    // When
+    val actual = mapper.toUpdateFutureWorkInterestsDto(request, prisonId)
+
+    // Then
+    assertThat(actual!!.prisonId).isEqualTo(prisonId)
+    assertThat(actual.reference).isEqualTo(request.id)
+    assertThat(actual.interests).usingRecursiveComparison().ignoringCollectionOrder().isEqualTo(expectedInterests)
+  }
+
+  @Test
+  fun `should map to UpdateFutureWorkInterestsDto when particular job interests is null`() {
+    // Given
+    val prisonId = "BXI"
+    val request = aValidUpdateWorkInterestsRequest(
+      workInterests = setOf(WorkType.TECHNICAL),
+      workInterestsOther = "",
+      particularJobInterests = emptySet(),
+    )
+    val expectedInterests = listOf(
+      WorkInterest(
+        workType = WorkInterestType.TECHNICAL,
+        workTypeOther = null,
+        role = null,
+      ),
+    )
+
+    // When
+    val actual = mapper.toUpdateFutureWorkInterestsDto(request, prisonId)
+
+    // Then
+    assertThat(actual!!.prisonId).isEqualTo(prisonId)
+    assertThat(actual.reference).isEqualTo(request.id)
+    assertThat(actual.interests).usingRecursiveComparison().ignoringCollectionOrder().isEqualTo(expectedInterests)
   }
```

Comment:
Do we need a test that aims to remove a job interest?

Reply:
I can only do that at the entity layer 🤯